File: ProjectOO/src/Requests/Offers.java
package Requests;

import Organization.Organization;

import java.util.ArrayList;
import java.util.stream.Collectors;

public class Offers extends RequestDonationList{
    public void commit(Organization organization) {
        for (RequestDonation i:rdEntities) {
            organization.getCurrentDonations().add(i);
        }
        rdEntities.clear();
    }
}

File: ProjectOO_C++/Offers.cpp
#include "Organization"
#include <stdio.h>
#include <vector>


class Offers: public RequestDonationList {
   public:
      void commit(Organization organization) {
        for (int i=0; i<rdEntities.size(); i++  ){
            organization.getCurrentDonations().add(rdEntities.at(i).getEntity().getId(),rdEntities.at(i).getQuantity());
        }
    }
};